app/services/mistral_provider.py

```
@@ -0,0 +1,130 @@
+        ]
+        
+        try:
+            results = await asyncio.gather(*tasks, return_exceptions=True)
+            translations = []
+            for result in results:
+                if isinstance(result, Exception):
+                    raise result
+                translations.append(result)
+            return translations
+        except Exception as e:
+            raise ProviderError(self.name, f"Batch translation failed: {str(e)}", e)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)